src/Common/Services/Mail/AggregateMailProvider.cs

```diff
@@ -5,53 +5,32 @@ namespace Passwordless.Common.Services.Mail;
 /// <summary>
 /// Wraps multiple mail providers and tries to send the message using them in order.
 /// </summary>
-public class AggregateMailProvider : IMailProvider
+public class AggregateMailProvider(
+    IOptionsSnapshot<MailConfiguration> options,
+    IMailProviderFactory factory,
+    ILogger<AggregateMailProvider> logger)
+    : IMailProvider
 {
-    private readonly IOptionsSnapshot<MailConfiguration> _options;
-    private readonly IMailProviderFactory _factory;
-    private readonly ILogger<AggregateMailProvider> _logger;
-
-    public const string FallBackFailedMessage = "No registered mail provider was able to send the message";
-
-    public AggregateMailProvider(
-        IOptionsSnapshot<MailConfiguration> options,
-        IMailProviderFactory factory,
-        ILogger<AggregateMailProvider> logger)
-    {
-        _options = options;
-        _factory = factory;
-        _logger = logger;
-    }
-
     public async Task SendAsync(MailMessage message)
     {
+        message.From ??= options.Value.From;
+        message.FromDisplayName ??= options.Value.FromName;
 
-        if (message.From == null)
-        {
-            message.From = _options.Value.From;
-        }
-        if (message.FromDisplayName == null)
-        {
-            message.FromDisplayName = _options.Value.FromName;
-        }
-        foreach (var providerConfiguration in _options.Value.Providers)
+        foreach (var providerConfiguration in options.Value.Providers)
         {
             try
             {
-                _logger.LogDebug("Attempting to send message using provider '{Provider}'", providerConfiguration.Name);
-                var provider = _factory.Create(providerConfiguration.Name, providerConfiguration);
+                logger.LogDebug("Attempting to send message using provider '{Provider}'", providerConfiguration.Name);
+                var provider = factory.Create(providerConfiguration.Name, providerConfiguration);
 
                 await provider.SendAsync(message);
-                _logger.LogInformation("Sent message using provider '{Provider}'", providerConfiguration.Name);
+                logger.LogInformation("Sent message using provider '{Provider}'", providerConfiguration.Name);
                 return;
             }
             catch (Exception e)
             {
-                _logger.LogError(e, "Failed to send message using provider '{Provider}'", providerConfiguration.Name);
+                logger.LogError(e, "Failed to send message using provider '{Provider}'", providerConfiguration.Name);
             }
         }
-
-        _logger.LogCritical(FallBackFailedMessage);
```

Comment:
If none of the registered e-mail providers can send, we need to know all of them fail. Because at that point the customers could be critically impacted being unable to log in.